server/src/controllers/baseLoad.ts

```diff
@@ -33,12 +34,18 @@ export const baseLoad: KoaController = async (ctx) => {
   await ORMContext.em.populate(user, ["save"]);
   const authSave = user.save;
   let save: Save = null;
+  logging(requestBody.type);
 
   if (requestBody.type === "build") {
     save = await loadBuildBase(ctx, requestBody.baseid);
     if(save && save.saveuserid !== user.userid) {
       throw saveFailureErr;
     }
+  } else if (requestBody.type === "ibuild") {
+    save = await loadBuildBase(ctx, authSave.baseid_inferno.toString());
+    if (save && save.saveuserid !== user.userid) {
+      throw saveFailureErr;
+    }
   } else {
     save = await loadViewBase(ctx, requestBody.baseid)
   }
```

Comment:
We could clean this up, not sure why we had this in the first place:
```javascript
let baseId = requestBody.baseid;
if (requestBody.type === "ibuild") {
baseId = authSave.baseid_inferno.toString()
}

save = await loadBuildBase(ctx, baseId);
if (save && save.saveuserid !== user.userid) {
  throw saveFailureErr;
}
```

If there is some reason we did not do error handling on non build requests we could just modify the last part to 
```javascript
if (requestBody.type.includes("build") && save && save.saveuserid !== user.userid) {
  throw saveFailureErr;
}
```

Reply:
Sorry let me fix comment above misread something